versioned/persist/tx-test/src/main/java/org/projectnessie/versioned/persist/tx/postgres/ContainerTestConnectionProviderSource.java

```diff
@@ -34,9 +36,23 @@ public void start() throws Exception {
       throw new IllegalStateException("Already started");
     }
 
-    container =
-        createContainer().withLogConsumer(new Slf4jLogConsumer(LOGGER)).withStartupAttempts(5);
-    container.start();
+    for (int retry = 0; ; retry++) {
+      container =
+          createContainer().withLogConsumer(new Slf4jLogConsumer(LOGGER)).withStartupAttempts(5);
+      try {
+        container.start();
+        break;
+      } catch (ContainerLaunchException e) {
+        container.close();
+        if (e.getCause() != null && e.getCause() instanceof ContainerFetchException && retry < 3) {
```

Comment:
Could we factor this code into a utility method and pass a lambda parameter from specific source implementations?

Reply:
The project don't have much in common that would be suitable for a utility class. Not sure, whether it's worth to add a new project just for this one.

Reply:
Meant the catch/retry logic, but the copy-paste is not terrible so +1